### PR TITLE
Bug fix/5704 validate run set name to be unique

### PR DESCRIPTION
**Pull request by vyasprateek**
Bug fix/5704 to restrict adding a Run Set with the name of existing run set .i.e., Validate run set name to be unique

### CODE REVIEW

File: Ginger/Ginger/RunSetLib/RunSetOperations.cs [hunk 1]
```diff
@@ -19,23 +19,46 @@ public static RunSetConfig CreateNewRunset(string runSetName="", RepositoryFolde
                 if (!InputBoxWindow.GetInputWithValidation(string.Format("Add New {0}", GingerDicser.GetTermResValue(eTermResKey.RunSet)), string.Format("{0} Name:", GingerDicser.GetTermResValue(eTermResKey.RunSet)), ref runSetName, System.IO.Path.GetInvalidPathChars()))
                 {
                     return null;
-                }                    
+                }
+
+                while (CheckAmbiguity(runSetName))
+                {
+                    Reporter.ToUser(eUserMsgKeys.ValueIssue, runSetName + " already exists ! Run Set name should be unique !");
+
```
Comment (MeniKadosh1) [inline Ginger/Ginger/RunSetLib/RunSetOperations.cs]: please add dedicated eUserMsgKeys for this message, and not need to use "!" in message, it should explain the issue in more pleseant manner.

File: Ginger/Ginger/RunSetLib/RunSetOperations.cs [hunk 2]
```diff
@@ -19,23 +19,46 @@ public static RunSetConfig CreateNewRunset(string runSetName="", RepositoryFolde
...
             else
             {
                 runSetsFolder.AddRepositoryItem(runSetConfig);
-            }                
+            }
 
             return runSetConfig;
         }
+
+        private static bool CheckAmbiguity(string runSetName)
```
Comment (MeniKadosh1) [inline Ginger/Ginger/RunSetLib/RunSetOperations.cs]: this method will work but you also could done it with simple Linq like:
if (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<RunSetConfig>().Where(x=>x.Name.ToLower()==NewName.ToLower()).FirstorNull() != null)

File: Ginger/Ginger/RunSetLib/RunSetOperations.cs [hunk 3]
```diff
@@ -37,21 +38,31 @@ public static RunSetConfig CreateNewRunset(string runSetName="", RepositoryFolde
                 if (!InputBoxWindow.GetInputWithValidation(string.Format("Add New {0}", GingerDicser.GetTermResValue(eTermResKey.RunSet)), string.Format("{0} Name:", GingerDicser.GetTermResValue(eTermResKey.RunSet)), ref runSetName, System.IO.Path.GetInvalidPathChars()))
                 {
                     return null;
-                }                    
+                }
+
+                while (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<RunSetConfig>().Where(r => r.ItemName.ToLower() == runSetName.ToLower()).FirstOrDefault() != null)
+                {
+                    Reporter.ToUser(eUserMsgKeys.DuplicateRunsetName, string.Format("'{0}' already exists, please use different name", runSetName));
```
Comment (MeniKadosh1) [inline Ginger/Ginger/RunSetLib/RunSetOperations.cs]: the actual message should be set in "UserMessagesPool" only the name of the Runset should be sent as parameter

File: Ginger/Ginger/RunSetLib/RunSetOperations.cs [hunk 4]
```diff
@@ -37,21 +38,31 @@ public static RunSetConfig CreateNewRunset(string runSetName="", RepositoryFolde
                 if (!InputBoxWindow.GetInputWithValidation(string.Format("Add New {0}", GingerDicser.GetTermResValue(eTermResKey.RunSet)), string.Format("{0} Name:", GingerDicser.GetTermResValue(eTermResKey.RunSet)), ref runSetName, System.IO.Path.GetInvalidPathChars()))
                 {
                     return null;
-                }                    
+                }
+
+                while (WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<RunSetConfig>().Where(r => r.ItemName.ToLower() == runSetName.ToLower()).FirstOrDefault() != null)
```
Comment (JinendraGhodke) [inline Ginger/Ginger/RunSetLib/RunSetOperations.cs]: No need of while if using Linq.

**Comment by MeniKadosh1** [comment]
Merged it and deleted the branch, please update the ticket

**Pull request merged**